cmd/sync-tags/main.go

```diff
@@ -258,71 +270,69 @@ func main() {
 		// update go.mod or Godeps.json to point to actual tagged version in the dependencies. This version might differ
 		// from the one currently in go.mod  or Godeps.json because the other repo could have gotten more commit for this
 		// tag, but this repo didn't. Compare https://github.com/kubernetes/publishing-bot/issues/12 for details.
+		var changed, goModExists bool
+		_, err = os.Stat("go.mod")
+		if err == nil {
+			goModExists = true
+		}
+
 		if len(dependentRepos) > 0 {
-			fmt.Printf("Checking that dependencies point to the actual tags in %s.\n", strings.Join(dependentRepos, ", "))
-			wt, err := r.Worktree()
-			if err != nil {
-				glog.Fatalf("Failed to get working tree: %v", err)
-			}
-			fmt.Printf("Checking out branch tag commit %s.\n", bh.String())
-			if err := wt.Checkout(&gogit.CheckoutOptions{Hash: bh}); err != nil {
-				glog.Fatalf("Failed to checkout %v: %v", bh, err)
-			}
+			wt := checkoutBranchTagCommit(r, bh, dependentRepos)
 
 			// if go.mod exists, fix only go.mod and generate Godeps.json from it later
 			// if it doesn't exist, check if Godeps.json exists, and update it
-			var changed, goModChanged bool
-			_, err = os.Stat("go.mod")
-			if os.IsNotExist(err) {
+			if !goModExists {
 				if _, err2 := os.Stat("Godeps/Godeps.json"); err2 == nil {
 					fmt.Printf("Updating Godeps.json to point to %s tag.\n", bName)
```

Comment:
should we replace Godeps.json with go.mod everywhere?

Reply:
or is this really still about godeps?

Reply:
This is for the case where there is no go.mod but only Godeps.json file in the repo. We don't need it for k8s repos but might be useful for non-k8s ones.

Or we could remove it here and force them to use the `godeps` branch in this repo, but that'd break compatibility.

Reply:
does this script still run on 1.14.x, or did we split the job to use the godeps branch for publishing that branch? dropping godeps manipulation seems ok, but I'd do it in a separate PR

Reply:
> does this script still run on 1.14.x, or did we split the job to use the godeps branch for publishing that branch?

The latter. Will remove this in another PR. :+1: